src/types/argument.js

```diff
@@ -17,6 +18,7 @@ export function initArgument(type) {
 
   switch (tag) {
     case GITypeTag.INTERFACE: {
+      // TODO: just generate a space large enough to hold the type
```

Comment:
we might be able to use `GObject.type_query` to get the `instance_size`, then we can generate just enough space to hold that interface like so:

```sh
return cast_ptr_u64(cast_buf_ptr(new ArrayBuffer(8 * instance_size));
```